source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -5,12 +5,14 @@
 using System.Collections;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
 using Newtonsoft.Json;
+using System.Threading;
 
 namespace AdaptiveCards.Test
 {
     [TestClass]
     public class AdaptiveCardApiTests
     {
+        List<Exception> exceptions = new List<Exception>();
```

Comment:
You can do away wiht the manual lock + List<> by switching to a ConcurrentBag<> which is thread safe and optimized for r/w from the calling thread.. I believe you dont care about the order of the exceptions in the list just that it should be non-empty - [https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/data-structures-for-parallel-programming](https://docs.microsoft.com/en-us/dotnet/standard/parallel-programming/data-structures-for-parallel-programming)

Reply:
thanks

---
In reply to: [340304521](https://github.com/microsoft/AdaptiveCards/pull/3546#discussion_r340304521) [](ancestors = 340304521)